code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -130,60 +177,111 @@
 			mode = BOT_IDLE
 			return
 
-	if(target && loc == target.loc)
-		clean(target)
-		path = list()
-		target = null
-
 	oldloc = loc
 
 /mob/living/simple_animal/bot/cleanbot/proc/get_targets()
-	target_types = new/list()
-
-	target_types += /obj/effect/decal/cleanable/blood/oil
-	target_types += /obj/effect/decal/cleanable/vomit
-	target_types += /obj/effect/decal/cleanable/blood/gibs/robot
-	target_types += /obj/effect/decal/cleanable/crayon
-	target_types += /obj/effect/decal/cleanable/liquid_fuel
-	target_types += /obj/effect/decal/cleanable/molten_object
-	target_types += /obj/effect/decal/cleanable/tomato_smudge
-	target_types += /obj/effect/decal/cleanable/egg_smudge
-	target_types += /obj/effect/decal/cleanable/pie_smudge
-	target_types += /obj/effect/decal/cleanable/flour
-	target_types += /obj/effect/decal/cleanable/ash
-	target_types += /obj/effect/decal/cleanable/greenglow
-	target_types += /obj/effect/decal/cleanable/dirt
+	target_types = list(
+		/obj/effect/decal/cleanable/blood/oil,
+		/obj/effect/decal/cleanable/vomit,
+		/obj/effect/decal/cleanable/robot_debris,
+		/obj/effect/decal/cleanable/molten_object,
+		/obj/effect/decal/cleanable/ash,
+		/obj/effect/decal/cleanable/greenglow,
+		/obj/effect/decal/cleanable/dirt,
+		/obj/effect/decal/cleanable/insectguts,
+		/obj/effect/decal/remains,
+		/obj/effect/decal/cleanable/flour,
+		/obj/effect/decal/cleanable/liquid_fuel,
+		/obj/effect/decal/cleanable/molten_object,
+		/obj/effect/decal/cleanable/tomato_smudge,
+		/obj/effect/decal/cleanable/egg_smudge,
+		/obj/effect/decal/cleanable/pie_smudge,
+		/obj/effect/decal/cleanable/blood/gibs/robot
+		)
 
 	if(blood)
-		target_types += /obj/effect/decal/cleanable/blood/xeno/
-		target_types += /obj/effect/decal/cleanable/blood/gibs/xeno
-		target_types += /obj/effect/decal/cleanable/blood/
-		target_types += /obj/effect/decal/cleanable/blood/gibs/
-		target_types += /obj/effect/decal/cleanable/blood/tracks
-		target_types += /obj/effect/decal/cleanable/dirt
+		target_types += /obj/effect/decal/cleanable/blood
 		target_types += /obj/effect/decal/cleanable/trail_holder
-
-/mob/living/simple_animal/bot/cleanbot/proc/clean(obj/effect/decal/cleanable/target)
-	anchored = 1
-	icon_state = "cleanbot-c"
-	visible_message("<span class='notice'>[src] begins to clean up [target]</span>")
-	mode = BOT_CLEANING
-	spawn(50)
-		if(mode == BOT_CLEANING)
-			QDEL_NULL(target)
+	
+	if(pests)
+		target_types += /mob/living/simple_animal/mouse
+
+	if(drawn)
+		target_types += /obj/effect/decal/cleanable/crayon
+
+	if(trash)
+		target_types += /obj/item/trash
+
+	target_types = typecacheof(target_types)
+
+/mob/living/simple_animal/bot/cleanbot/UnarmedAttack(atom/A, proximity_flag, list/modifiers)
```

Comment:
I feel it's better to keep the logic below here in the `clean` proc. Calling `UnarmedAttack` while you want to clean is a bit odd. So it's best to keep the old construction of having `UnarmedAttack` call `clean` and `clean` being called by the bot logic

Reply:
I ended up changing a lot more, I hope it makes more sense this way.